exporter/exportertest/sink_exporter.go

```diff
@@ -20,14 +20,52 @@ import (
 
 	"github.com/open-telemetry/opentelemetry-collector/component"
 	"github.com/open-telemetry/opentelemetry-collector/consumer/consumerdata"
+	"github.com/open-telemetry/opentelemetry-collector/internal/data"
 )
 
-// SinkTraceExporter acts as a trace receiver for use in tests.
-type SinkTraceExporter struct {
+// SinkTraceExporterOld acts as a trace receiver for use in tests.
+type SinkTraceExporterOld struct {
 	mu     sync.Mutex
 	traces []consumerdata.TraceData
 }
 
+// Start tells the exporter to start. The exporter may prepare for exporting
+// by connecting to the endpoint. Host parameter can be used for communicating
+// with the host after Start() has already returned.
+func (ste *SinkTraceExporterOld) Start(host component.Host) error {
+	return nil
+}
+
```

Comment:
Instead of adding all these new code to all the classes I would do the following:

```
type traceExporterWrapper() struct {
	te *TraceExporter
}

// All functions that are required to implement TraceExporteOld just redirect into te, and for consume data we do the conversion then call into te.

NewTraceExporterOldWrapper(te *TraceExporter) *TraceExporterOld {
  return &traceExporterWrapper{te};
}

```

What do you think?

Reply:
I wouldn't merge them, just to keep it easy to throw away `SinkTraceExporterOld` when it's not needed

Reply:
I think the wrappers are useful not only for this class but to do faster transition. It is much easier in contrib to just use the wrappers and convert everyone to the new model.

Reply:
I gave this approach a try. But looks like it introduces more complexity than benefit we are gaining. We can reuse `Start` and `Shutdown` methods only, which are no-op essentially. The methods consuming traces/metrics should be kept separate for both classes, because clients of old and new consumers need their own data formats. Or do I miss something?